integration/pvtdata/data_purge_test.go

```diff
@@ -8,151 +8,248 @@ package pvtdata
 
 import (
 	"context"
+	"encoding/base64"
+	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"syscall"
 
 	"github.com/hyperledger/fabric/integration/nwo"
+	"github.com/hyperledger/fabric/integration/nwo/commands"
 	"github.com/hyperledger/fabric/integration/pvtdata/marblechaincodeutil"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/tedsuo/ifrit"
 )
 
 var _ = Describe("Pvtdata purge", func() {
-	var (
-		testDir   string
-		network   *nwo.Network
-		orderer   *nwo.Orderer
-		org2Peer0 *nwo.Peer
-		process   ifrit.Process
-		cancel    context.CancelFunc
-		chaincode *nwo.Chaincode
-	)
-
-	BeforeEach(func() {
-		var err error
-		testDir, err = ioutil.TempDir("", "purgedata")
-		Expect(err).NotTo(HaveOccurred())
-
-		config := nwo.ThreeOrgRaft()
-		network = nwo.New(config, testDir, nil, StartPort(), components)
-
-		network.GenerateConfigTree()
-		network.Bootstrap()
-
-		networkRunner := network.NetworkGroupRunner()
-		process = ifrit.Invoke(networkRunner)
-		Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
-
-		orderer = network.Orderer("orderer")
-		network.CreateAndJoinChannel(orderer, "testchannel")
-		network.UpdateChannelAnchors(orderer, "testchannel")
-		network.VerifyMembership(
-			network.PeersWithChannel("testchannel"),
-			"testchannel",
+	When("the purge private data capability is not enabled", func() {
```

Comment:
This `When` could be placed after `BeforeEach` and `AfterEach`. So, in some sense, the common code for defining variables, network setup, and tear down should be defined once before any `When`. This would remove the duplicate code below as well.

Reply:
The reason for the When was to have separate `BeforeEach`s since the network is very slightly different (purge is not enabled in one, and enabled in the other. From memory Matt Sykes expected more code duplication in test code so I didn't worry about it but happy to investigate alternatives for setting up the network differently with more common code.